### PR TITLE
Check that main_ids are str or int

### PR DESCRIPTION
**Pull request by alejoe91**
Came up in #2053. We always assume channel/unit ids are str or ints, but we have no explicit check for it in main. 

This PR fixes it.

### CODE REVIEW

File: src/spikeinterface/extractors/cellexplorersortingextractor.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def __init__(
         spike_times = spikes_data["times"]
 
         # CellExplorer reports spike times in units seconds; SpikeExtractors uses time units of sampling frames
-        unit_ids = unit_ids[:].tolist()
+        unit_ids = unit_ids[:].astype(int).tolist()
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/cellexplorersortingextractor.py:121]: Should we do strings to start moving in that direction?
```suggestion
        unit_ids = unit_ids[:].astype(int).tolist()
        unit_ids = [str(unit_id) for unit_id in unit_ids]
```

File: src/spikeinterface/extractors/cellexplorersortingextractor.py [hunk 2]
```diff
@@ -118,7 +118,8 @@ def __init__(
...
+        unit_ids = [str(unit_id) for unit_id in unit_ids]
```
Comment (samuelgarcia) [inline src/spikeinterface/extractors/cellexplorersortingextractor.py:122]: Why are we going to str in this extractor ?
Why do we need int and then str and why not directly str ?

Comment (alejoe91) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py:122]: right, we only need the second line

Comment (alejoe91) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py:121]: done

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**